sample/becat.c

```diff
@@ -188,6 +188,10 @@ static void ssl_ctx_free(struct ssl_context *ssl)
 static int ssl_load_key(struct ssl_context *ssl)
 {
 	int err = 1;
+#if OPENSSL_VERSION_MAJOR >= 3
```

Comment:
I don't think OpenSSL 1.x defines OPENSSL_VERSION_MAJOR - you might need
OPENSSL_VERSION_NUMBER >= 0x30000000

Reply:
Indeed, nice catch.

Also CI reports:

```
2022-06-23T08:16:11.2428545Z ../sample/le-proxy.c:116:5: warning: "OPENSSL_VERSION_MAJOR" is not defined, evaluates to 0 [-Wundef]
2022-06-23T08:16:11.2429120Z  #if OPENSSL_VERSION_MAJOR >= 3
```

I will add `-Werror` on CI - #1297

Reply:
Fixed.